great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -52,7 +52,8 @@ class SerializableDataContext(AbstractDataContext):
         DataContextConfigDefaults.PROFILERS_BASE_DIRECTORY.value,
         GX_UNCOMMITTED_DIR,
     ]
-    GX_DIR: ClassVar[str] = "great_expectations"
+    GX_DIR: ClassVar[str] = "gx"
+    _LEGACY_GX_DIR: ClassVar[str] = "great_expectations"
```

Comment:
https://github.com/great-expectations/great_expectations/pull/8420 ensures this change will be seamless (as long as we've used constants across the codebase) - docs will need to be updated though